app/uk/gov/hmrc/gform/auth/models/JwtPayload.scala

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.gform.auth.models
+
+import play.api.libs.functional.syntax._
+import play.api.libs.json.Reads._
+import play.api.libs.json.{ JsPath, Reads }
+
+/*
+  These set of fields are common to the JWT coming from the public LB and the admin LB
+ */
+case class JwtPayload(
+  sub: String,
+  username: String,
+  exp: Long,
+  iss: String
+)
+
+object JwtPayload {
+  implicit val format: Reads[JwtPayload] = (
```

Comment:
Why is this `Reads` not derived?

Reply:
What do you mean? Can you expand?

Reply:
Something like this:
```
implicit val format = Json.format[JwtPayload]
```
Here the macro `format` is automatically generating the same code for us, and we don't need to maintain unnecessary boilerplate.

Reply:
Yes, it was like that before. But the issue now is that we have two types of users that can come through the Ofsted route. Strangely the JWT for these two different types of users is different. However, they do have common fields and they are represented in this structure. So when I parse the JSON and try to create an instance of JwtPayload I only want to read the common fields. It saves me creating two case classes: AdminJwtPayload and UserJwtPayload.

Reply:
Derived `Reads` will read nothing else that what fields name of `JwtPayload` are. Is that wrong?

Reply:
No not really. It's just habit. I can change it.

Reply:
I've changed it to use derived reads.